### ISSUE TITLE
[bcachefs-tools] can't resize fs

### ISSUE THREAD

**Issue by marcin-github** [koverstreet/bcachefs#39]
```
# bcachefs device resize /dev/crypted/dane2
Doing offline resize of /dev/crypted/dane2
journal read done, 33 keys in 1 entries, seq 286244
starting metadata mark and sweep
starting mark and sweep
starting fsck
mounted with opts: compression=zstd
resizing /dev/crypted/dane2 to 163840 buckets
bcachefs: libbcachefs/journal.c:976: bch2_fs_journal_stop: Assertion `!(!bch2_journal_error(j) && test_bit(JOURNAL_NOT_EMPTY, &j->flags))' failed.
Aborted (core dumped)
```

```
# echo bt | gdb -q /sbin/bcachefs 'core-1557213154-0-6-!sbin!bcachefs-13787'
Reading symbols from /sbin/bcachefs...(no debugging symbols found)...done.
[New LWP 13787]
[New LWP 13788]
[New LWP 13794]
[New LWP 13789]
[New LWP 13797]
[New LWP 13791]
[New LWP 13812]
[New LWP 13796]
[New LWP 13793]
[New LWP 13795]
[New LWP 13790]
[New LWP 13792]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `bcachefs device resize /dev/crypted/dane2'.
Program terminated with signal SIGABRT, Aborted.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50        return ret;
[Current thread is 1 (Thread 0x7fc2b19aad80 (LWP 13787))]
(gdb) #0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fc2affe27a5 in __GI_abort () at abort.c:79
#2  0x00007fc2affe267f in __assert_fail_base (fmt=<optimized out>, assertion=0x558957cce7bf "!(!bch2_journal_error(j) && test_bit(JOURNAL_NOT_EMPTY, &j->flags))",
    file=0x558957cce5da "libbcachefs/journal.c", line=976, function=0x558957ccea70 "bch2_fs_journal_stop") at assert.c:92
#3  0x00007fc2afff0f22 in __GI___assert_fail (assertion=0x558957cce7bf "!(!bch2_journal_error(j) && test_bit(JOURNAL_NOT_EMPTY, &j->flags))", file=0x558957cce5da "libbcachefs/journal.c",
    line=976, function=0x558957ccea70 "bch2_fs_journal_stop") at assert.c:101
#4  0x0000558957c87f5b in ?? ()
#5  0x0000558957c98060 in ?? ()
#6  0x0000558957c9844e in ?? ()
#7  0x0000558957c98864 in ?? ()
#8  0x0000558957cbb42a in ?? ()
#9  0x0000558957c49cd2 in ?? ()
#10 0x00007fc2affe469b in __libc_start_main (main=0x558957c49a70, argc=4, argv=0x7ffec0114b58, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffec0114b48)
    at ../csu/libc-start.c:308
#11 0x0000558957c4a75a in ?? ()
(gdb) quit
```

**Comment by marcin-github** [comment]
Argh, this is because filesystem was mounted (I remember about problem with detecting).

**Comment by nicman23** [comment]
yes #22

**Comment by koverstreet** [comment]
you actually did hit a real bug - fixed by 507d33d9d0

**Comment by marcin-github** [comment]
Thanks!
